examples/builder_derive_interop/2_augment_subcommands.rs

```diff
@@ -0,0 +1,35 @@
+use clap::{Arg, Command, Parser, Subcommand as _};
+
+#[derive(Parser)]
+enum Subcommands {
+    Derived {
+        #[clap(short, long)]
+        derived_flag: bool,
+    },
+}
+
+fn main() {
+    let cli = Command::new("Built CLI").subcommand(
+        Command::new("built").arg(Arg::new("built-flag").short('b').long("built-flag")),
+    );
+    // Augment built subcommands with derived subcommands
+    let cli = Subcommands::augment_subcommands(cli);
+
+    let matches = cli.get_matches();
+    match matches.subcommand() {
+        Some(("built", sub_matches)) => {
+            println!(
+                "Got built subcommand with flag {:?}",
+                sub_matches.is_present("built-flag")
+            );
+        }
+        Some(("derived", sub_matches)) => {
+            println!(
+                "Got derived subcommand with flag {:?}",
+                sub_matches.is_present("derived-flag")
+            );
+        }
+        None => println!("No subcommand"),
+        _ => unreachable!(),
+    }
```

Comment:
Like with `Args`, we should be showing how to use `FromArgMatches` for `Subcommands`.  This has a consequence that we should not be showing mixing builder subcommands with derive subcommands.  That should be a less common case and so an exercise for the reader to infer form this.

Reply:
How do we handle the built subcommand? Should I just remove it?

Reply:
Yes, I think it should be removed.